src/resolvedVariables.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+/**
+ * Resolve variables aliases.
+ */
+module.exports = function resolvedVariables(ctx) {
+  var cache = {};
+
+  ctx.data
+    .filter(isVariable)
+    .forEach(function (item) {
+      cache[item.context.name] = item.context.value;
+    });
+
+  for (var item in cache) {
+    var value = variableValue(cache[item]);
+
+    if (value) {
+      cache[item] = cache[value[1]];
+    }
+  }
+
+  ctx.data
+    .filter(isVariable)
+    .forEach(function (item) {
+      item.resolvedValue = cache[item.context.name];
+    });
+};
+
+/**
+ * Test whether passed item is of type variable.
+ * @param {Object} item
+ * @return {Boolean}
+ */
+function isVariable(item) {
+  return typeof item.context.type === 'string' &&
+         item.context.type.toLowerCase() === 'variable';
+}
+
+/**
+ * Test and extract variable as item value.
+ * @param {String} value
+ * @return {Array | null}
+ */
+function variableValue(value) {
+  return /^\$([a-zA-Z-_.]+)/.exec(value);
```

Comment:
We might encounter some edge-case issues with variables containing escaped characters.